e2e/scripts/fund-e2e-accounts.ts

```diff
@@ -30,24 +41,27 @@ const valoraTestFaucetSecret = process.env['TEST_FAUCET_SECRET']!
   const ceurExchange = await kit.contracts.getExchange(StableToken.cEUR)
 
   // Balance Faucet
-  let targetTokenRegex = /[cusd|celo|ceur]/gi
-  let balances = await getBalance(E2E_TEST_FAUCET)
-  let sum = 0
-  const numOfTokens = 3 // Only cusd, celo and ceur
-  for (let balance in balances) {
-    console.log(`${balance}: ${balances[balance]}`)
-    // Sum only the target balances
-    if (targetTokenRegex.test(balance)) sum += balances[balance]
-  }
+  let totalTokenHoldings = 0 // the absolute number of faucet tokens the faucet is holding
+  Object.entries(faucetTokenBalances).forEach(([tokenSymbol, tokenBalance]) => {
+    if (REFILL_TOKENS.includes(tokenSymbol)) {
+      totalTokenHoldings += tokenBalance
+    }
+  })
+  const targetFaucetTokenBalance = totalTokenHoldings / REFILL_TOKENS.length
 
-  for (let balance in balances) {
-    const target = sum / numOfTokens
-    if (balances[balance] >= sum / numOfTokens) {
-      const toSell = balances[balance] - target
-      console.log(toSell)
-      const amountToExchange = kit.web3.utils.toWei(`${toSell}`, 'ether')
-      console.log(`${balance} balance higher than ${sum / numOfTokens}`)
-      switch (balance) {
+  // Ensure that the faucet has enough balance for each refill tokens
+  for (const [tokenSymbol, tokenBalance] of Object.entries(faucetTokenBalances)) {
+    if (!REFILL_TOKENS.includes(tokenSymbol)) {
```

Comment:
not strictly necessary, but without this we'd proceed for tokens like cREAL and TT only to hit the default case in the switch statement (no transaction created)